src/esi/Parser.cc

```diff
@@ -22,8 +22,11 @@ ESIParser::NewParser(ESIParserClient *aClient)
     if (Parser == NULL) {
         Parser = Parsers;
 
-        while (Parser != NULL && strcasecmp(Parser->name, Type) != 0)
-            Parser = Parser->next;
+        // if type name matters, use it
+        if (strcasecmp(Type, "auto") == 0) {
```

Comment:
Mnemonic: `==` means `the strings are equal`. You want `!=` here AFAICT.

Reply:
Thanks. fixed